#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/drinks/protos/Can.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/fruits/protos/Apple.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/fruits/protos/Orange.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/kitchen/breakfast/protos/JamJar.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/kitchen/breakfast/protos/CerealBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/obstacles/protos/OilBarrel.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/kitchen/breakfast/protos/BiscuitBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/toys/protos/RubberDuck.proto"

WorldInfo {
  info [
    "Example use of a color Camera device with recognition capability."
  ]
  title "Camera Image Recognition"
}
Viewpoint {
  orientation 0.3593668066583357 -0.027767781981431972 -0.9327831733881257 3.0249769164763864
  position 2.31346691628409 0.3564303580705131 1.9856188541136333
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 1.5 1.5
}
Can {
  translation 0.267626 0.290201 0.0609617108609355
  rotation 0.6078229636147192 0.7886071931714658 0.09300505244793011 4.863450869639638e-17
}
Can {
  translation -0.326272 -0.341474 0.0609617108609355
  rotation 0.6078229636147192 0.7886071931714658 0.09300505244793011 4.863450869639638e-17
  name "can(1)"
}
Apple {
  translation -0.260886 0.22946 0.049764791394518226
}
Apple {
  translation -0.621889 0.56252 0.049764791394518226
  name "apple(1)"
}
DEF ORANGE Solid {
  translation -0.002188 -0.063045 0
  children [
    Orange {
      translation -0.187654 0.182317 0.04999764793150722
    }
  ]
  name "orange"
  recognitionColors [
    0.988235 0.686275 0.243137
  ]
}
DEF ORANGE Solid {
  translation 0.085807 -0.503005 0
  children [
    Orange {
      translation -0.187654 0.182317 0.04999764793150722
    }
  ]
  name "orange(1)"
  recognitionColors [
    0.988235 0.686275 0.243137
  ]
}
JamJar {
  translation -0.33681 0.078775 0
  mass 0
}
JamJar {
  translation 0.314033 -0.288298 0
  rotation 0 0 1 2.094395
  name "jam jar(1)"
  mass 0
}
CerealBox {
  translation -0.43865 0.194464 -0.00039201630959942446
  rotation 3.770103787649246e-17 2.407949098820323e-18 -1 0.523599
}
CerealBox {
  translation -0.45025899348281107 0.41529999585090943 -0.00039220694597741357
  rotation -7.438330797950161e-08 -2.0088280332138234e-06 -0.9999999999979794 0.7853985634923097
  name "cereal box(1)"
}
OilBarrel {
  translation -0.28712 -0.211927 0.07
  rotation 0 0 1 -1.5707953071795862
  height 0.14
  radius 0.06
}
OilBarrel {
  translation -0.240466 -0.485864 0.07
  rotation 0 0 1 -1.5707953071795862
  name "oil barrel(1)"
  height 0.14
  radius 0.06
}
OilBarrel {
  translation -0.464787 -0.437183 0.07
  rotation 0 0 1 -1.5707953071795862
  name "oil barrel(2)"
  height 0.14
  radius 0.06
}
BiscuitBox {
  translation 0.5384121770599435 -0.27466816841412345 0.03984308800740199
  rotation 0.6947462352857393 -0.18615610946793437 -0.694747127711986 -2.773493129695837
}
RubberDuck {
  translation -0.243817 0.384468 0.020436910949152134
  rotation -2.1881405658049035e-17 -2.92732624931207e-17 1 -2.356195307179586
}
Robot {
  translation 0.08410562623689305 0.0014106288729985607 -0.00023419623914434917
  rotation -0.014148600228101087 -0.02242242633020867 0.9996484641658044 -1.126111857642492
  children [
    Pose {
      translation 0 0 0.0415
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -6.287722767639882
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.045 0.025
        rotation 0 1 0 1.2939778467192906
        children [
          DEF WHEEL Pose {
            rotation -1 0 0 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        linearVelocity -0.01420493695935347 0.030671649320537887 1.1144860676707066e-09
        angularVelocity -1.3620879523949871 -0.628500977682968 0.7569269184885055
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 6.2876966453003496
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.04499999999999999 0.025
        rotation 0 1 0 1.7003466453003497
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
        linearVelocity 0.014205166070513888 -0.030937244470759125 -2.184152883453608e-08
        angularVelocity 1.3620851582192308 0.6284992543711816 0.7569274975035941
      }
    }
    DistanceSensor {
      translation 0.042 0.02 0.063
      rotation 0 0 -1 -0.499205
      children [
        DEF SENSOR Pose {
          rotation 0 1 0 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.042 -0.02 0.063
      rotation 0 0 1 -0.500795
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/samples/devices/worlds/textures/mybot.png"
          ]
        }
        roughness 0.5
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0.0435 -0.015 0.038
            0.0435 -0.015 0.023
            0.0455 0 0.023
            0.0435 0.015 0.023
            0.0435 0.015 0.038
            0.0455 0 0.038
          ]
        }
        texCoord TextureCoordinate {
          point [
            0 0
            0.5 0
            1 0
            1 1
            0.5 1
            0 1
          ]
        }
        coordIndex [
          0, 1, 2, 5, -1, 5, 2, 3, 4, -1
        ]
        texCoordIndex [
          5, 0, 1, 4, -1, 4, 1, 2, 3, -1
        ]
        creaseAngle 0.9
      }
    }
    Camera {
      translation 0.04 0 0.0915
      children [
        Pose {
          translation -0.015 0 0
          rotation 0 -1 0 1.57
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.03
                radius 0.01
              }
            }
          ]
        }
      ]
      fieldOfView 1.0472
      width 256
      height 128
      antiAliasing TRUE
      recognition Recognition {
        frameColor 0.929412 0.831373 0
        frameThickness 3
        segmentation TRUE
      }
    }
    Display {
      name "segmented image display"
      width 256
      height 128
    }
  ]
  name "MyBot"
  boundingObject Pose {
    translation 0 0 0.0415
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "camera_segmentation_py"
  linearVelocity 0.0003196654091316393 1.4865879239021946e-05 -6.017788899739916e-08
  angularVelocity -1.4031926523922842e-06 -8.543114049760736e-07 0.7528223685022579
}
